code/modules/clothing/masks/miscellaneous.dm

```diff
@@ -18,7 +24,43 @@
 /obj/item/clothing/mask/muzzle/attack_hand(mob/user as mob)
 	if(user.wear_mask == src && !user.IsAdvancedToolUser())
 		return 0
+	else if(user.wear_mask == src && security_lock && locked)
+		if(ishuman(user))
+			if(doUnlock(user))
+				visible_message("<span class='danger'>[user] unlocks their [src.name].</span>", \
+									"<span class='userdanger'>[user] unlocks their [src.name].</span>")
+		else
+			to_chat(user, "<span class='warning'>You lack the ability to open the lock.</span>")
+			return 0
 	..()
+	return 1
+
+/obj/item/clothing/mask/muzzle/proc/doUnlock(mob/living/carbon/human/user)

```

Comment:
This whole proc is a mess. 
First off, please use underscores rather than camelcase, to be consistent with the few other masks that have their own procs. 
Secondly, you shouldn't be using `user.get_idcard` when all you want to do is check if they have access- use `user.get_access()` instead. If there was any event where this gets called with a silicon user, it would matter otherwise, but it's all typecasted to humans.
Thirdly, the proc would be better formatted as this:
```
/obj/item/clothing/mask/muzzle/proc/do_unlock(mob/living/carbon/human/user)
	var/unlock = FALSE

	if(istype(user.get_inactive_hand(), /obj/item/weapon/card/emag))
		to_chat(user, "<span class='warning'>The lock vibrates as the card forces its locking system open.</span>")
		security_lock = FALSE
		unlock = TRUE

	else if(access_brig in user.get_access())
		unlock = TRUE

	if(unlock)
		to_chat(user, "<span class='warning'>The muzzle unlocks with a click.</span>")
		locked = FALSE
		flags &= ~NODROP
		return TRUE

	to_chat(user, "<span class='warning'>You must be wearing a security ID card or have one in your inactive hand to remove the muzzle.</span>")
	return FALSE
```